File: app/src/main/java/com/monk/retrofit/base/BasePresent.java
package com.monk.retrofit.base;

/**
 * @author monk
 * @date 2019-02-19
 */
public class BasePresent <T extends IBaseView>{
    protected T mMvpView;

    public BasePresent(T mMvpView) {
        this.mMvpView = mMvpView;
    }

    public T getMvpView() {
        return mMvpView;
    }

    public void detachMvpView(){
        mMvpView=null;
    }
}

File: module_fragment/src/main/java/com/monk/modulefragment/retrofit/base/BasePresent.kt
package com.monk.modulefragment.retrofit.base

/**
 * @author monk
 * @date 2019-02-19
 */
open class BasePresent<T : IBaseView?>(mMvpView: T) {
    var mvpView: T?
        protected set

    fun detachMvpView() {
        mvpView = null
    }

    init {
        mvpView = mMvpView
    }
}